src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -0,0 +1,496 @@
+  autoPlayActive: boolean;
+  startDate: moment.Moment;
+  endDate: moment.Moment;
+}
+
+/**
+ * mapStateToProps - mapping state to props of Map Component.
+ *
+ * @param {Object} state current state
+ * @return {Object} mapped props
+ */
+
+/**
+ * The panel combining all time slider related parts.
+ */
+export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelProps, TimeLayerSliderPanelState> {
+
+  private _TimeLayerAwareSlider: any;
+  private _wmsTimeLayers: any[];
+  private _interval: number;
+
+  /**
+   * The default props of LayerSetBaseMapChooser
+   *
+   * @static
+   * @type {DefaultLayerSetBaseMapChooserProps}
+   * @memberof LayerSetBaseMapChooser
+   */
+  public static defaultProps: DefaultTimeLayerSliderPanelProps = {
+    className: '',
+    onChange: () => {/* empty */},
+    timeAwareLayers: [],
+    value: moment(moment.now()),
+    dateFormat: 'YYYY-MM-DD',
+    tooltips: {
+      setToNow: 'Set to now',
+      hours: 'Hours',
+      days: 'Days',
+      weeks: 'Weeks',
+      months: 'Months',
+      years: 'Years',
+      dataRange: 'Set data range'
+    }
+  };
+
+  /**
+   * Constructs time panel.
+   */
+  constructor(props: TimeLayerSliderPanelProps) {
+    super(props);
+
+    this.state = {
+      value: moment().milliseconds(0),
+      playbackSpeed: '1',
+      autoPlayActive: false,
+      startDate: moment().milliseconds(0),
+      endDate: moment().milliseconds(0),
+    };
+    this._interval = 1000;
+
+    this.wrapTimeSlider();
+
+    // binds
+    this.onTimeChanged = this.onTimeChanged.bind(this);
+    this.autoPlay = this.autoPlay.bind(this);
+    this.updateDataRange = this.updateDataRange.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      initStartDate,
+      initEndDate
+    } = this.props;
+
+    this.setState({
+      startDate: initStartDate,
+      endDate: initEndDate
+    });
+  }
+
+  componentDidUpdate(prevProps: TimeLayerSliderPanelProps) {
+    // TODO: this deep check may impact performance..
+    if (!(_isEqual(prevProps.timeAwareLayers, this.props.timeAwareLayers))) {
+      // update slider properties if some another layer set was chosen
+      this.wrapTimeSlider();
+      this.findRangeForLayers();
+    }
+  }
+
+  /**
+   *
+   * @param nextProps
+   * @param nextState
+   */
+  shouldComponentUpdate(nextProps: TimeLayerSliderPanelProps, nextState: TimeLayerSliderPanelState) {
+    const {
+      value,
+      autoPlayActive,
+      startDate,
+      endDate
+    } = this.state;
+    const {
+      timeAwareLayers
+    } = this.props;
+
+    if (nextState.value !== value) {
+      return true;
+    }
+    if (nextState.autoPlayActive !== autoPlayActive) {
+      return true;
+    }
+    if (nextState.startDate !== startDate) {
+      return true;
+    }
+    if (nextState.endDate !== endDate) {
+      return true;
+    }
+    if (!(_isEqual(nextProps.timeAwareLayers, timeAwareLayers))) {
+      return true;
+    }
+    return false;
```

Comment:
I like the approach, but please also add update support for at least the `tooltips` prop.